src/web/controllers/admin/index_admin_controller.py

```diff
@@ -403,52 +307,409 @@ async def htmx_admin_tab(
+                rules = stored_event.rules
+                message_text = stored_event.message_text
+                message_color = admin_event.message_color
+                message_background_color = admin_event.message_background_color
+                prize_currency = stored_event.prize_currency
+                override_unrated_rapid_blitz = stored_event.override_unrated_rapid_blitz
+            case 'create':
+                sharly_chess_config: SharlyChessConfig = SharlyChessConfig()
+                public = False
+                background_color = sharly_chess_config.default_background_color
+                message_background_color = (
+                    sharly_chess_config.default_message_background_color
+                )
+                message_color = sharly_chess_config.default_message_color
+                federation = (
+                    sharly_chess_config.federation.name
+                    if sharly_chess_config.federation
+                    else ''
+                )
+                hide_background_image = (
+                    sharly_chess_config.default_hide_background_image
+                )
+                override_unrated_rapid_blitz = True
+            case _:
+                raise ValueError(f'action=[{action}]')
+
+        per_plugin_form_data = plugin_manager.hook.get_event_form_data(
+            event=admin_event
+        )
+        plugin_form_data = {
+            key: value for data in per_plugin_form_data for key, value in data.items()
+        }
+
+        return {
+            'uniq_id': WebContext.value_to_form_data(uniq_id),
+            'name': WebContext.value_to_form_data(name),
+            'public': WebContext.value_to_form_data(public),
+            'federation': WebContext.value_to_form_data(federation),
+            'start': WebContext.value_to_datetime_form_data(start),
+            'stop': WebContext.value_to_datetime_form_data(stop),
+            'background_image_checkbox': WebContext.value_to_form_data(
+                hide_background_image
+            ),
+            'background_image': WebContext.value_to_form_data(background_image),
+            'background_color': WebContext.value_to_form_data(background_color),
+            'location': WebContext.value_to_form_data(location),
+            'record_illegal_moves': WebContext.value_to_form_data(record_illegal_moves),
+            'rules': WebContext.value_to_form_data(rules),
+            'message_text': WebContext.value_to_form_data(message_text),
+            'message_color': WebContext.value_to_form_data(message_color),
+            'message_background_color': WebContext.value_to_form_data(
+                message_background_color
+            ),
+            'prize_currency': WebContext.value_to_form_data(prize_currency),
+            'override_unrated_rapid_blitz': WebContext.value_to_form_data(
+                override_unrated_rapid_blitz
+            ),
+        } | plugin_form_data
```

Comment:
Note: Youri is probably the best at saying if it is worth doing this.
```suggestion
        return WebContext.values_dict_to_form_data(
            {
                'uniq_id': uniq_id,
                'name': name,
                'public': public,
                'federation': federation,
                'background_image_checkbox': hide_background_image,
                'background_image': background_image,
                'background_color': background_color,
                'location': location,
                'record_illegal_moves': record_illegal_moves,
                'rules': rules,
                'message_text': message_text,
                'message_color': message_color,
                'message_background_color': message_background_color,
                'prize_currency': prize_currency,
                'override_unrated_rapid_blitz': override_unrated_rapid_blitz,
            }
        ) | {
            'start': WebContext.value_to_datetime_form_data(start),
            'stop': WebContext.value_to_datetime_form_data(stop),
        } | plugin_form_data
```

Reply:
That doesn't work for me